ai/default/aiunit.cpp

```diff
@@ -796,7 +797,9 @@ int look_for_charge(struct ai_type *ait, struct player *pplayer,
         continue;
       }
       // Reduce want based on move cost.
-      def >>= move_cost / (2 * std::min(1, unit_move_rate(punit)));
+      int divisor = (2 * std::min(1, unit_move_rate(punit)));
```

Comment:
This looks like a logic error, the `min` will always return 1. Should likely be a `max`.

Reply:
You are probably right, but I'd rather not change existing logic, when the goal is to fix a possible division by zero reported by a static analyzer.

Reply:
The only reason that could cause a division by zero is this error in the first place :P